core/src/stores/postgres.rs

```diff
@@ -1748,24 +1761,16 @@ impl Store for PostgresStore {
     async fn upsert_data_source_document(
         &self,
         project: &Project,
-        data_source_id: &str,
-        document: &Document,
-        title: Option<String>,
-        mime_type: Option<String>,
-    ) -> Result<()> {
+        data_source_id: String,
+        upsert_params: DocumentUpsertParams,
+        create_params: Option<DocumentCreateParams>,
+    ) -> Result<Document> {
+        let document_created = match create_params {
```

Comment:
`created` does not get replaced on update. Only set on insert. To make this crispy clear I put it in that separate `create_params` struct.

Reply:
Discussed IRL there is no Upsert here :magic_wand:

Reply:
fixed !